### PR TITLE
bump to kubernetes 1.17.0-rc.2

### PR DESCRIPTION
**Pull request by mfojtik**

### CODE REVIEW

File: glide.yaml [hunk 1]
```diff
@@ -1,20 +1,24 @@
...
 import:
-  # this matches the 1.16 branch from kube
-  - package: k8s.io/gengo
-    version: 26a664648505d962332bda642b27306bc10d1082
-  - package: k8s.io/code-generator
-    repo:    https://github.com/openshift/kubernetes-code-generator.git
-    version: client-go-4.3-kubernetes-1.16.0-beta.1
   - package: k8s.io/apimachinery
-    version: kubernetes-1.16.0
+    version: kubernetes-1.17.0-rc.2
```
Comment (deads2k) [inline glide.yaml]: 1.17.0

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,18 @@
+module github.com/openshift/client-go
+
+go 1.13
+
+require (
+	github.com/openshift/api v3.9.1-0.20191213091414-3fbf6bcf78e8+incompatible
+	k8s.io/api v0.17.0
+	k8s.io/apimachinery v0.17.1-beta.0
```
Comment (deads2k) [inline go.mod]: 0.17.0

File: go.mod [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+	k8s.io/client-go v0.0.0-20191204083517-ea72ff2b5b2f
```
Comment (deads2k) [inline go.mod]: v0.17.0

File: go.mod [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+)
+
+replace (
+	github.com/openshift/api => github.com/openshift/api v0.0.0-20191213091414-3fbf6bcf78e8
+	k8s.io/api => k8s.io/api v0.0.0-20191210230129-4c9a86741a7a
```
Comment (deads2k) [inline go.mod]: k8s.io/api, apimachinery, client-go are extraneous

File: hack/update-codegen.sh [hunk 5]
```diff
@@ -9,8 +9,14 @@ CODEGEN_PKG=${CODEGEN_PKG:-$(cd ${SCRIPT_ROOT}; ls -d -1 ./vendor/k8s.io/code-ge
...
 
+# Because go mod sux, we have to fake the vendor for generator in order to be able to build it...
+mv ${CODEGEN_PKG}/generate-groups.sh ${CODEGEN_PKG}/generate-groups.sh.orig
+sed 's/ go install/#go install/g' ${CODEGEN_PKG}/generate-groups.sh.orig > ${CODEGEN_PKG}/generate-groups.sh
+go install ./${CODEGEN_PKG}/cmd/{defaulter-gen,client-gen,lister-gen,informer-gen,deepcopy-gen}
+mv ${CODEGEN_PKG}/generate-groups.sh.orig ${CODEGEN_PKG}/generate-groups.sh
+
 for group in apps authorization build config console image imageregistry network oauth project quota route samples security template user; do
-  ${CODEGEN_PKG}/generate-groups.sh "client,lister,informer" \
+  bash ${CODEGEN_PKG}/generate-groups.sh "client,lister,informer" \
```
Comment (deads2k) [inline hack/update-codegen.sh]: why?

Comment (mfojtik) [reply hack/update-codegen.sh]: go mod remove executable bit...

**Comment by deads2k** [comment]
the input is horrible.  the output is tolerable.  

/lgtm

**Pull request merged**